src/Symfony/Component/Messenger/Exception/ThrownExceptionDetails.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Messenger\Exception;
+
+use Symfony\Component\ErrorHandler\Exception\FlattenException;
+
+/**
+ * ThrownExceptionDetails captures details of a thrown exception and is displayed like a normal exception when dumped
+ * with the VarDumper.
+ *
+ * @author Jeroen Noten <jeroennoten@me.com>
+ */
+class ThrownExceptionDetails
+{
+    private $class;
+    /**
+     * @var string
+     */
+    private $message;
+    /**
+     * @var int
+     */
+    private $code;
+    /**
+     * @var string
+     */
+    private $file;
+    /**
+     * @var int
+     */
+    private $line;
+    /**
+     * @var array
+     */
+    private $trace;
+
+    public function __construct(
+        string $class,
+        string $message,
+        int $code,
+        string $file,
+        int $line,
+        array $trace
+    ) {
+        $this->class = $class;
+        $this->message = $message;
+        $this->code = $code;
+        $this->file = $file;
+        $this->line = $line;
+        $this->trace = $trace;
+    }
+
+    public static function createFromFlattenException(FlattenException $flattenException): self
+    {
+        return new self(
+            $flattenException->getClass(),
+            $flattenException->getMessage(),
+            $flattenException->getCode(),
+            $flattenException->getFile(),
+            $flattenException->getLine(),
+            $flattenException->getTrace()
+        );
+    }
```

Comment:
Let's don't use the constructor with plenty of arguments. 

```suggestion
    public static function createFromFlattenException(FlattenException $flattenException): self
    {
        $model = new self();
        $model->class = $flattenException->getClass();
        $model->message = $flattenException->getMessage();
        $model->code = $flattenException->getCode();
        $model->file = $flattenException->getFile();
        $model->line = $flattenException->getLine();
        $model->trace = $flattenException->getTrace();

        return $model;
    }
```

Reply:
Done!